### PR TITLE
Remove the method proxy magic

### PR DESCRIPTION
**Pull request by hackaugusto**
This change forces the code to be explicit about doing a transaction or
a call, and removes the magic used in the proxy constructor to set its
attributes

### CODE REVIEW

File: raiden/network/rpc/smartcontract_proxy.py [hunk 1]
```diff
@@ -2,75 +2,47 @@
...
-                address,
-                function_name,
-                self.translator,
-                call_func,
-                transact_func,
-                estimate_function,
-            )
 
-            type_argument = self.translator.function_data[function_name]['signature']
+VALID_KARGS = {'gasprice', 'startgas', 'value'}
```
Comment (palango) [inline raiden/network/rpc/smartcontract_proxy.py]: Is this used somewhere else? Otherwise it should be a static member

Comment (palango) [reply raiden/network/rpc/smartcontract_proxy.py]: I there any advantage of the global over the old approach?: https://github.com/hackaugusto/raiden/blob/ca2564bb/raiden/network/rpc/smartcontract_proxy.py#L50

Comment (hackaugusto) [reply raiden/network/rpc/smartcontract_proxy.py]: Not that I can think of.

Comment (palango) [reply raiden/network/rpc/smartcontract_proxy.py]: Let's switch this to the class member approach then. Then this PR is good to go.

**Pull request merged**